packages/framework/src/errors/base.errors.ts

```diff
@@ -33,8 +36,27 @@ export abstract class UnauthorizedError extends FrameworkError {
   statusCode = HttpStatusEnum.UNAUTHORIZED;
 }
 
-export abstract class InternalServerError extends FrameworkError {
-  statusCode = HttpStatusEnum.INTERNAL_SERVER_ERROR;
+export abstract class ServerError extends FrameworkError {
+  data: {
+    /**
+     * The stack trace of the error.
+     */
+    stack: string;
+  };
+
+  constructor(...[message, { cause }]: DeepRequired<ConstructorParameters<typeof Error>>) {
+    if (isNativeError(cause)) {
+      super(`${message}: ${cause.message}`);
+      this.data = {
+        stack: cause.stack ?? message,
+      };
+    } else {
+      super(`${message}: ${JSON.stringify(cause, null, 2)}`);
+      this.data = {
+        stack: message,
+      };
+    }
+  }
```

Comment:
A base server error that encapsulates handling of stack-traces onto the `data` object.